elastalert/elastalert.py

```diff
@@ -456,8 +456,21 @@ def get_hits_count(self, rule, starttime, endtime, index):
             five=rule['five']
         )
 
+        es_client = self.thread_data.current_es
         try:
-            res = self.thread_data.current_es.count(index=index, doc_type=rule['doc_type'], body=query, ignore_unavailable=True)
+            if es_client.is_atleastsixsix():
+                res = self.thread_data.current_es.count(
+                    index=index,
+                    body=query,
+                    ignore_unavailable=True
+                )
+            else:
+                res = self.thread_data.current_es.count(
```

Comment:
With `doc_type`:

```
self.thread_data.current_es.count(
                    index=index,
                    doc_type=rule['doc_type'],
                    body={'query': {'match_all': {}}},
                    ignore_unavailable=True
                )
{'count': 0, '_shards': {'total': 1, 'successful': 1, 'skipped': 0, 'failed': 0}}
```

Reply:
This is silently just not matching any documents in 7.x.  :(

Reply:
Looks to match if you use the implicit `_doc` type, but this makes side by side migrations harder where 6.x clusters had a `doc_type` different from the default.